src/window.py

```diff
@@ -183,27 +182,23 @@ def _on_activate_link(self, _, uri):
             return True
         return False
 
-    def _update_header(self, state):
-        self.audio_video_switcher_wdg.set_visible(state == 'start')
-        self.download_wdg.set_visible(state == 'start')
-        self.back_wdg.set_visible(state in ['success', 'error'])
-        self.cancel_wdg.set_visible(state in ['download', 'cancel'])
-        self.title_wdg.set_visible(state != 'start')
-
     def _update_focus_and_default(self, _):
         state = self.main_stack_wdg.get_visible_child_name()
         mode = self.audio_video_stack_wdg.get_visible_child_name()
         if state == 'start':
-            self.download_wdg.grab_default()
             if mode == 'audio':
+                self.audio_download_wdg.grab_default()
                 self.audio_url_wdg.grab_focus()
             elif mode == 'video':
+                self.video_download_wdg.grab_default()
                 self.video_url_wdg.grab_focus()
             else:
                 assert False
-        elif state in ['download', 'cancel']:
-            self.cancel_wdg.grab_focus()
-        elif state in ['success', 'error']:
-            self.back_wdg.grab_focus()
+        elif state == 'download':
```

Comment:
This should be
```python3
state in ['download', 'cancel']
```

Reply:
I didn't understand what the cancel state is, as there is no child named like that in the UI file and so the state variable should never hold this value. Is that child added somewhere and I didn't see it? Or maybe is the download page's name changed somehow?

Reply:
The cancel state uses the download UI child:

https://github.com/Unrud/video-downloader/blob/eb12092c31bf1051d297363c7404669c081bb21e/src/window.py#L67-L69